File: Javascript/Basic Hashing/Problem2.js

//Find the highest/lowest frequency element


let arr=[10,5,10,15,10,5];


let map=new Map();

for(let i=0;i<arr.length;i++){

  if(map.has(arr[i])){
    let getval=map.get(arr[i]);
    map.set(arr[i],getval+1);
  }
  else{
    map.set(arr[i],1);
  }
}

let max=-Infinity;
let min=Infinity;
let maxval;
let minval;
for(let [key,value] of map){
  
  if(max<value){
    max=value;
    maxval=key;
  }
  
   if(min>value){
    min=value;
    minval=key;
  }
 
 
}
console.log(max,maxval,min,minval)

File: C++/Basic Hashing/Problem2.cpp
//Find the highest/lowest frequency element

#include <iostream>
#include <bits/stdc++.h>
using namespace std;

int main() 
{
   int arr[]={10,5,10,15,10,5};
   
   int length=sizeof(arr)/sizeof(arr[0]);
   
   unordered_map<int,int>mapp;
   for(int i=0;i<length;i++){
     mapp[arr[i]]++;
   }
   int max=0;
   int maxval;
   int min=length;
   int minval;
   
   for(auto it : mapp){
     if(max<it.second){
       max=it.second;
       maxval=it.first;
     }
      if(min>it.second){
       min=it.second;
       minval=it.first;
     }
     
   }
   std::cout << maxval<<" "<<minval << std::endl;
    return 0;
}